src/lib/sidebar/ListMode.svelte

```diff
@@ -40,6 +40,56 @@
       errorFromFile = `Couldn't load scheme from a file: ${err}`;
     }
   }
+
+  function moveSchemeInList(scheme_reference: string, direction: number) {
+    return function () {
+      const currentSchemeOrder = JSON.parse(
+        JSON.stringify(Object.keys($gjSchemeCollection.schemes))
+      );
+      const currentIndex = currentSchemeOrder.indexOf(scheme_reference);
+      if (
+        currentIndex + direction >= 0 &&
+        currentIndex + direction < currentSchemeOrder.length
+      ) {
+        const newSchemeOrder = swapArrayValues(
+          currentSchemeOrder,
+          currentIndex,
+          currentIndex + direction
+        );
+        const newSchemesObject = getReorderedSchemesObject(
+          newSchemeOrder,
+          $gjSchemeCollection.schemes
+        );
+        $gjSchemeCollection.schemes = newSchemesObject;
+        $gjSchemeCollection = $gjSchemeCollection;
+      }
+    };
+  }
+
+  function swapArrayValues(
+    array: any[],
+    firstIndex: number,
+    secondIndex: number
+  ) {
+    const firstValue = array[firstIndex];
+    const secondValue = array[secondIndex];
+    array[secondIndex] = firstValue;
+    array[firstIndex] = secondValue;
+
+    return array;
+  }
+
+  function getReorderedSchemesObject(
```

Comment:
All of this works fine for me in modern Firefox, but according to MDN, `Object.keys` does not guarantee any kind of ordering. https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order is a minefield; I couldn't find a canonical newer answer (preferably in someplace like MDN). We should maaaybe use something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames instead, but it seems to work now, so let's just go with it.